LSPS4/README.md

```diff
@@ -0,0 +1,1788 @@
+  * This is done by checking for the existence of a
+    "PPP Opening Set", described later.
+* MUST check that the given `opening_fee_params.promise` validates
+  the given `opening_fee_params`, and that
+  `opening_fee_params.valid_until` is a future datetime.
+* MUST check that all the PPPs listed in `ppps` are still in the
+  PPP Map.
+* MUST sum up the amounts of all PPPs listed and set `payment_size`
+  to that sum.
+* MUST compute the `opening_fee` based on the `payment_size` and
+  the given `opening_fee_params`, and MUST check that the
+  `opening_fee` is strictly less than
+  `payment_size - number_of_ppps * client_htlc_minimum_msat`.
+
+If all above checks pass, the call succeeds.
+Before responding, the LSP, still in the same atomic section:
+
+* Removes the PPPs listed in `ppps` from the PPP Map and puts
+  them into a PPP Opening Set.
+  * The existence of the PPP Opening Set is what is checked to
+    determine if a pending LSPS4-bought channel is being opened.
+
+Once the above operations complete, the call succeeds and the
+atomic section ends, and the LSP responds with an empty object
+`{}` and proceeds to the next step.
+
+If the checks fail, the atomic section ends and the LSP responds
+with an error code:
+
+* `operation_pending` (2) - An LSPS4-bought channel is not yet
+  completed opening and sending the PPPs being used to pay for
+  it.
+  * The client SHOULD retry later once the channel has opened and
+    the existing payments resolved.
+* `invalid_opening_fee_params` (3) - The `opening_fee_params.promise`
+  failed validation, or the `opening_fee_params.valid_until` is
+  already a past datetime.
+* `nonexistent_ppp` (4) - One or more of the `ppps` is no longer in
+  the PPP Map.
+  The error `data` object contains the required field
+  `unknown_ppps`, which is an array of PPP identifiers
+  containing the PPPs which were not found.
+* `insufficient_funds` (5) - The total `payment_size` is not
+  enough to pay for the `opening_fee`.
+
+> **Non-normative** The intent is that once asynchronous receives
+> are possible, asyncpay-type PPPs will NOT be accepted in the
+> `lsps4.open_with_ppps` call.
+> Instead, those PPPs can only be passed to `lsps4.forward_ppps`
+> call, which will cause the LSP to respond to the senders of those
+> asynchronous receives.
+> The senders would then deliver actual HTLCs/PTLCs that the LSP
+> will convert again to PPPs.
+>
+> This will help with the multiple-small-separate-payments
+> case; asyncpay-type PPPs are expected to be holdable by the
+> LSP longer, and once enough asyncpay-type PPPs are available,
+> the client can then request the senders to try sending the
+> actual payments in the hope that the LSP can gather enough
+> to pay for a new channel.
+
+### 7. Opening Channel And Forwarding Selected Pending Payment Parts
+
+The LSP requests a channel open to the client via standard
+[BOLT2 Channel Establishment][] flow.
+
+The LSP selects the channel size.
```

Comment:
Yes, that is why the LSP has this decision and why `opening_fee` is based on payment size.  This was already decided on in LSPS2.

Reply:
We decided how we calculate the fee. We never talked about who determines the channel size as far as I know.

Reply:
https://github.com/BitcoinAndLightningLayerSpecs/lsp/pull/22#discussion_r1163172128

Reply:
Basically:

* Channel size is a cost on the LSP: the larger the channel size, the greater the cost on the LSP (needs more UTXOs, risks locking up more of their liquidity to a user that ends up not using it, etc).
* LSP thus wants cost of JIT channel to be proportional to the capacity of the channel it will make.
* UX is harder if the client software decides channel size by itself without human input.
* Human only cares about payment size, for UX we want client software to hide the channel size decisionmaking from the human.
* Payment size is approximately an estimate on how much capacity the human will need in the future; it is likely that the human will receive more inflow of similar size in the future.
* Thus: human indicates payment size to client software, LSP charges based on payment size (good for UX), LSP now has cost basis for how large a channel it can build to the client.

Reply:
Cool. Let's try this then. In case there is demand/good reason for client determined channel size, we can always ship a v2.